### PR TITLE
rename Compose -> Semicategory, and make Compose the Haskell one 

### PR DESCRIPTION
**Pull request by hrhino**
Following up on a gitter discussion. Multiple commits for history preservation purposes.

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/data/compose.scala [hunk 1]
```diff
@@ -0,0 +1,185 @@
+package scalaz
+package data
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/data/compose.scala]: Someone else please tell me if this is `data` or `ct` or something else. I can't tell and don't really care to puzzle it out.

File: example/src/main/tut/ct/Semigroupoid.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+layout: docs
+title:  "Semigroupoid"
+---
+
+# Semigroupoid [![GitHub](../img/github.png)](https://github.com/scalaz/scalaz/blob/series/8.0.x/base/shared/src/main/scala/scalaz/ct/compose.scala)
```
Comment (NeQuissimus) [inline example/src/main/tut/ct/Semigroupoid.md]: The link to the sources needs to change

Comment (hrhino) [reply example/src/main/tut/ct/Semigroupoid.md]: Good catch!

File: base/shared/src/main/scala/scalaz/data/compose.scala (hunk 1 above)
Comment (jdegoes) [reply base/shared/src/main/scala/scalaz/data/compose.scala]: `ct`

**Comment by hrhino** [comment]
On Tue, Jun 12, 2018 at 10:20 AM, John A. De Goes
<notifications@github.com> wrote:
>
> @jdegoes commented on this pull request.
>
> ________________________________
>
> In base/shared/src/main/scala/scalaz/data/compose.scala:
>
> > @@ -0,0 +1,185 @@
> +package scalaz
> +package data
>
> ct
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Why?

File: base/shared/src/main/scala/scalaz/data/compose.scala (hunk 1 above)
Comment (hrhino) [reply base/shared/src/main/scala/scalaz/data/compose.scala]: Alrighty then.

File: example/src/main/tut/ct/Compose.md [hunk 3]
```diff
@@ -1,6 +1,23 @@
 ---
 layout: docs
-title:  "Compose"
+title:  "Forall"
```
Comment (alexknvl) [inline example/src/main/tut/ct/Compose.md]: `Forall` => `Compose`

**Comment by tonymorris** [comment]
The `Compose` type was written before the haskell `semigroupoids` package, hence the naming differences. Renaming this to `Semigroupoid` has long been on the todo list.

However, note that there is a current desire to rename this to Semicategory in haskell.
https://github.com/ekmett/semigroupoids/issues/26

**Comment by hrhino** [comment]
Okay, I opportunistically changed it to `Semicategory` as I trust nLab over wikipedia. @emilypi is there a reason, other than ekmett, to call it `Semigroupoid`?

**Comment by hrhino** [comment]
So my main goal here is to have this `Compose`.

I don't really care what the not-quite-`Category` thing is called. `Semicategory` doesn't sound awful to me.

If a representative of the oid-rage gang speaks up I'm more than happy to change it back. Otherwise I'll merge this tomorrow or later this week. Obviously anyone can feel free to rename it later.

File: base/shared/src/main/scala/scalaz/ct/semicategory.scala [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package scalaz
+package ct
+
+import scala.language.experimental.macros
+
+trait SemigroupoidClass[=>:[_, _]] {
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/ct/semicategory.scala]: ah heck, I'm sloppy

- [ ] pick one you fool

**Comment by emilypi** [comment]
I'm actually okay with considering it a [Semicategory](https://ncatlab.org/nlab/show/semicategory).

**Comment by hrhino** [comment]
Aight, a `Semicategory` it shall be.

**Pull request merged**